### PR TITLE
EZP-29801: Content View breaks when providing location id as param

### PR DESCRIPTION
**Pull request by andrerom**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-29801](https://jira.ez.no/browse/EZP-29801)
| **Bug/Improvement**| yes/no
| **New feature**    | no
| **Target version** | `7.2+`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

If you somehow provide location id to Content View, then it will throw the following now after ContentProxy feature was added in 2.2:
```
Argument 'Location' is invalid: Provided location does not belong to selected content as requested via contentId parameter
```

This for instance break Step 3 in Bike Ride tutorial where you are asked to enter following url, last parameter is optional so removing it makes it work:
`http://<yourdomains>/view/content/<ContentId>/<language>/full/true/<LocationId>`

Culprit is the following code which is wrongly using strict check:
```php
if (isset($parameters['contentId']) && $location->contentId !== $parameters['contentId']) {
```

**TODO**:
- [x] Implement feature / fix a bug.
- [x] ~Implement tests.~
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ public function buildView(array $parameters)
                 throw new InvalidArgumentException('Location', 'Provided location does not belong to selected content');
             }
 
-            if (isset($parameters['contentId']) && $location->contentId !== $parameters['contentId']) {
+            if (isset($parameters['contentId']) && $location->contentId != $parameters['contentId']) {
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php]: Please cast it to `int` instead of doing weak comparison.

Comment (andrerom) [reply eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php]: fixed in 46790a2

**Comment by emodric** [comment]
Late to discussion here, but, wouldn't it be better to fix the cause (i.e. whatever is setting the ID to be a `string` instead of `int`) and not the manifestation? It's not hard to imagine other places where this issue might popup.

**Comment by andrerom** [comment]
We could try to see how we can get Symony to pass us parameters as int in those cases. But this was just regression fix here so did not look into it. Did you?

**Comment by emodric** [comment]
No, not yet :) But good enough for now ;)

**Pull request merged**